src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx

```diff
@@ -0,0 +1,386 @@
+        })}
+      </EuiButton>
+    );
+  };
+
+  const renderToolsRight = () => {
+    if (selection.length === 0) {
+      return;
+    }
+    return (
+      <Actions
+        permissionSettings={permissionSettings}
+        isTableAction={false}
+        selection={selection}
+        handleSubmitPermissionSettings={handleSubmitPermissionSettings}
+      />
+    );
+  };
+
+  const search: EuiSearchBarProps = {
+    box: {
+      incremental: true,
+    },
+    filters: [
+      {
+        type: 'field_value_selection',
+        field: 'displayedType',
+        name: 'Type',
+        multiSelect: false,
+        options: Array.from(new Set(items.map(({ displayedType }) => displayedType ?? ''))).map(
+          (item) => ({
+            value: item,
+            name: item,
+          })
+        ),
+      },
+      {
+        type: 'field_value_selection',
+        field: 'accessLevel',
+        name: 'Access level',
+        multiSelect: false,
+        options: Array.from(new Set(items.map(({ accessLevel }) => accessLevel ?? ''))).map(
+          (item) => ({
+            value: item,
+            name: item,
+          })
+        ),
+      },
+    ],
+    toolsLeft: renderToolsLeft(),
+    toolsRight: renderToolsRight(),
+  };
+
+  const columns: Array<EuiBasicTableColumn<PermissionSetting>> = [
+    {
+      field: 'primaryId',
+      name: 'ID',
+    },
+    {
+      field: 'displayedType',
+      name: 'Type',
+    },
+    {
+      field: 'accessLevel',
+      name: 'Access level',
+    },
+    {
+      name: 'Actions',
+      field: '',
+      render: (item: PermissionSetting) => (
+        <Actions
+          isTableAction={true}
+          selection={[item]}
+          permissionSettings={permissionSettings}
+          handleSubmitPermissionSettings={handleSubmitPermissionSettings}
+          openDeleteConfirmModal={openDeleteConfirmModal}
+        />
+      ),
+    },
+  ];
+  const selectionValue: EuiTableSelectionType<PermissionSetting> = {
+    onSelectionChange: (newSelection) => setSelection(newSelection),
+  };
+
+  return (
+    <EuiInMemoryTable
+      items={items}
+      columns={columns}
+      search={search}
+      itemId="id"
+      pagination={true}
+      message={emptyStateMessage}
+      isSelectable={true}
+      selection={selectionValue}
+    />
+  );
+};
+
+const Actions = ({
+  isTableAction,
+  selection,
+  permissionSettings,
+  handleSubmitPermissionSettings,
+  openDeleteConfirmModal,
+}: {
+  isTableAction: boolean;
+  selection?: PermissionSetting[];
+  permissionSettings: PermissionSetting[];
+  handleSubmitPermissionSettings: (permissionSettings: WorkspacePermissionSetting[]) => void;
+  openDeleteConfirmModal?: ({ onConfirm }: { onConfirm: () => void }) => { close: () => void };
+}) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const { overlays } = useOpenSearchDashboards();
+
+  const accessLevelOptions = Object.keys(WORKSPACE_ACCESS_LEVEL_NAMES).map((level) => ({
```

Comment:
Nit: We can use `Object.keys(WORKSPACE_ACCESS_LEVEL_NAMES) as WorkspaceCollaboratorAccessLevel[]`
, then we don't need too many `as WorkspaceCollaboratorAccessLevel` below.